hibernate-core/src/main/java/org/hibernate/LockOptions.java

```diff
@@ -363,4 +371,14 @@ else if ( this == that ) {
 		return followOnLocking != null ? followOnLocking.equals( that.followOnLocking ) : that.followOnLocking == null;
 	}
 
+	@Override
+	public int hashCode() {
+		int result = lockMode != null ? lockMode.hashCode() : 0;
+		result = 31 * result + timeout;
+		result = 31 * result + ( aliasSpecificLockModes != null ? aliasSpecificLockModes.hashCode() : 0 );
+		result = 31 * result + ( followOnLocking != null ? followOnLocking.hashCode() : 0 );
+		result = 31 * result + ( scope ? 1 : 0 );
+		return result;
+	}
```

Comment:
It would be great to use **Objects** static util methods to simplify, if only from code readability perspective.